### ISSUE TITLE
MySQL Connector 8.2.0 issue

### ISSUE THREAD

**Issue by Charles-II** [dbeaver/dbeaver#32940]
### Description

I'm using DBeaver to access MySQL InnoDB clusters version 8.0.36. DBeaver version 22.3.5 with mysql-8.0.31-win64 driver and mysql-connector-j-8.0.29 is working well and I can get the DB table list in few seconds (using filter to reduce table amount). If I upgrade it to newer versions I can open the connection and directly run the queries but as soon as I try to get the filtered table list (through Databases/Tables menu item) a "Load child Items" process starts and the memory occupation grows until the maximum allowed (I have increased it up to 5GB) and then I get "SQL Error [S1000] Java heap Space" error and DBeaver stops working. I think that the issue is related to the new mysql-connector-j-8.2.0 which forcefully replaces the older one. I have also tried to update the mysql driver to version 8.2.0 without success.

[DBeaver24.0.4Error.log](https://github.com/dbeaver/dbeaver/files/15319888/DBeaver24.0.4Error.log)

### DBeaver Version

Community Edition 24.0.4 and 23.3.5

### Operating System

Windows 11 

### Database and driver

Driver mysql-8.0.31-win64
Connector mysql-connector-j-8.2.0

### Steps to reproduce

Expand Tables item even with filter applied

### Additional context

No problems with DBeaver version 22.3.5, mysql connector mysql-connector-j-8.0.29 and driver mysql-8.0.31-win64
The issue starts with DBeaver 23.x.x up to latest one (all using new mysql-connector-j-8.2.0)

**Comment by MashaKorax** [comment]
Did you try to use the same version of the  mysql-connector-j-8.0.29 and driver mysql-8.0.31-win64?

**Comment by MashaKorax** [comment]
Please make screenshot of libriries 
![iScreen Shoter - DBeaver Community - 240516150352](https://github.com/dbeaver/dbeaver/assets/84867187/0a64cfba-1f7a-4795-bf6c-0c882505f6cf)

**Comment by Charles-II** [comment]
I tried to keep the previous mysql-connector-j-8.0.29 but the DBeaver 24.0.4 install procedure forced me to replace it. If you could explain to me how to install it with the latest DBeaver version I will do a try.

**Comment by Charles-II** [comment]
![Connector8 2 0_Info](https://github.com/dbeaver/dbeaver/assets/66421564/549a175c-5f0b-4752-bfbb-9105809b1320)
Using DBeaver from version 23.x.x up to 24.0.4

**Comment by Charles-II** [comment]
![Connector8 0 29](https://github.com/dbeaver/dbeaver/assets/66421564/11ebad0a-c45d-4442-a4f9-a523a0d191e2)
Using DBeaver 22.3.5 (last version which is working well)

**Comment by Charles-II** [comment]
![MySQL8_driver](https://github.com/dbeaver/dbeaver/assets/66421564/37d4c4b1-fefe-45ee-b34a-d6000c3d7ed8)
MySQL driver version common to all DBeaver versions.

**Comment by E1izabeth** [comment]
Please try decreasing the driver version
![image](https://github.com/dbeaver/dbeaver/assets/28875055/be1e0a33-c07a-45bc-85c7-b9bc04726210)

**Comment by Charles-II** [comment]
I tried but the problem is still there. I have also tried with the last 24.0.5 version with both connector versions and the problem persists. The last working version is still 22.3.5
EDIT: I have to say that Dbeaver version 22.3.5 uses mysql-connector-j-8.0.29 but with newer DBeaver versions I cannot install connector versions earlier than 8.0.31. So there is still a difference with the connector.

**Comment by E1izabeth** [comment]
@Charles-II 
Does it reproducible without filtering?

**Comment by E1izabeth** [comment]
How many tables do you have in this database?
Have you tried changing -Xmx1024m into -Xmx4096m to increase maximum memory to 4Gb in dbeaver.ini file?

**Comment by Charles-II** [comment]
Yes, the problem happens with and without filtering. As reported in the first post I have increased the maximum memory up to 5Gb and the problem is still there. Using the task monitor I can see the memory occupation growing up to the maximum and then I get Java Heap Space error. DBeaver version 22.3.5 works perfectly with and without filtering out tables and even with a 1Gb of maximum memory.

**Comment by Charles-II** [comment]
I would like to add that I'm not the only one facing this problem. The entire team which is working on the same DB has faced the same issue and we are all forced to use DBeaver version 22.3.5. to avoid it.
We are using MySQL InnoDB version 8.0.36 with 234071 tables, 58 tables have more than 1GB of data of which 3 tables have data length of 110GB (72MB index), 226GB (3GB index) and 1207GB (19GB index) .

**Comment by Charles-II** [comment]
The problem is still present with version 24.1.4

**Comment by vifelisberto** [comment]
EDIT:

I managed to solve it by removing the mysql and lgpl libraries and downloading the dependency directly from the website: https://repo1.maven.org/maven2/com/mysql/mysql-connector-j/8.2.0/

I created a folder called com.mysql inside .local/share/DBeaverData/drivers/maven/maven-central and added the file inside (mysql-connector-j-8.2.0.jar)

Then I went to the Dbeaver Driver Manager -> Mysql -> Edit -> Libraries -> Add File and selected the jar file

After that I clicked on Find Class and selected com.mysql.jbdc.Driver and confirmed the screens

I don't know if this is the ideal option, but it started working again.

SO: Linux Mint
Dbeaver 24.3.0
